lib/SetAPI/generic/SetInterfaceV1.py

```diff
@@ -89,12 +131,82 @@ def _check_save_set_params(self: "SetInterfaceV1", params: dict[str, Any]) -> No
             if not params.get(param):
                 raise ValueError(param_required(param))
 
+        # this should not happen with the current API, but best to check
+        if not set_item_name or set_item_name not in SET_ITEM_NAME_TO_SET_TYPE:
+            err_msg = f"invalid set item name: {set_item_name}"
+            raise ValueError(err_msg)
+
+        # bail out early for RNASeqSampleSets
+        if set_item_name == RNASEQ_SAMPLE:
+            return set_item_name
```

Comment:
sample sets are first processed by the SampleSetInterface, which does various bits of parameter coercion. The results of that are handed to the SetInterfaceV1 to do the standard saving rigmarole. Because the sample structure within the save parameters differs from all the other set types, we don't do any further checks.